doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+# Practical Overview of Explainability
+
+While the applications of machine learning are impressive many models provide predictions that are hard to interpret or reason about. This limits their use in many cases as often we want to know why and not just what a model made a prediction. Alarming predictions are rarely taken at face value and typically warrant further analysis. Indeed, in some
+cases explaining the choices that a model makes could even become a potential [legal requirement](https://arxiv.org/pdf/1711.00399.pdf). The following is a non-rigorous and practical overview of explainability and the methods alibi provide.
+
+**Explainability provides us with algorithms that give insights into trained models predictions.** It allows
+us to answer questions such as:
+- How does a prediction change dependent on feature inputs?
+- What features are Important for a given prediction to hold?
+- What features are not important for a given prediction to hold?
+- What set of features would you have to minimally change to obtain a new prediction of your choosing?
+- How does each feature contribute to a model's prediction?
+
+The set of insights available are dependent on the trained model. For instance, If the model is a regression it makes sense to ask how the prediction varies for some feature. Whereas, it doesn't make sense to ask what minimal change is required to obtain a new classification. Insights are constrained by:
+
+- The type of data the model handles. Some insights only apply to image data others only to textual data
+- The task the model performs. The two types of model tasks alibi handles are regression and classification
+- The type of model used. Examples of model types include neural networks and random forests
+
+Some explainer methods apply only to specific types of models such as TreeSHAP which can only be used with tree-based models. This is the case when an explainer method uses some aspect of that model's structure. If the model is a neural network then some methods require computing the prediction derivative for model inputs. Methods that require access to the model internals like this are known as **white box** methods. Other explainers apply to any type of model. They can do so because the underlying method doesn't make use of the model internals. Instead, they only need to have access to the model outputs given particular inputs. Methods that apply in this general setting are known as **black box** methods. Note that white box methods are a subset of black-box methods and an explainer being a white box method is a much stronger constraint than it being a black box method. Typically, white box methods are faster than black-box methods as access to the model internals means the method can exploit some aspect of that model's structure.
```

Comment:
structure --> internal structure ?
"prediction derivative for model inputs" -> "prediction gradient with respect to model inputs"
"like this" --> " "
"white box" --> "white-box" (globally)
"black box" --> "black-box" (globally)

Reply:
I would even change "computing the prediction derivative for model inputs" to "taking gradients of the model predictions with respect to the inputs".